styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java

```diff
@@ -62,4 +72,46 @@ public FlyteExecution createExecution(final String name, final FlyteExecConf fly
       throw new CreateExecutionException(flyteExecConf, e);
     }
   }
+
+  public void poll(final String project, final String domain, final String name, RunState runState) {
+    Objects.requireNonNull(project);
+    Objects.requireNonNull(domain);
+    Objects.requireNonNull(name);
+    final ExecutionOuterClass.Execution execution =
+        flyteAdminClient.getExecution(project, domain, name);
```

Comment:
Probably we need to catch exception here, unless it would be handled in the upper layers

Reply:
https://github.com/spotify/styx/blob/master/styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java#L94
RuntimeExceptions are not handled in the equivalent layer of the current impl, so i thought it was probably fine to do the same here

Reply:
I was looking at this and I think that the reason that `DockerRunnerHandler.java` is not handling exception is because [KubernetesDockerRunner](https://github.com/spotify/styx/blob/9ecb3f967ff86a0fe36c29d82b7d4026b679c6cd/styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java#L107) is doing so. So it is not because it is handled un the upper layer but because it is handled in the lower layer. We should do the same as Styx shouldn't crash because a flyteadmin connection issue

Reply:
yes i thought it's better to let the poll method handle the exception. Better also to keep everything in one place.
It seems that the KubDockerRunner is ignoring that one. But i think it's better to handle it explicitely.